src/client/testing/testController/pytest/pytestDiscoveryAdapter.ts

```diff
@@ -98,6 +119,20 @@ export class PytestTestDiscoveryAdapter implements ITestDiscoveryAdapter {
         };
         const execService = await executionFactory?.createActivatedEnvironment(creationOptions);
         // delete UUID following entire discovery finishing.
+        /*
+        here we are putting together all the args for the pytest discovery.
+        THIS SECTION I AM OPEN TO SUGGESTIONS, i am not sure if my suggested flow below is the best so please let me know if you have a better idea.
+
+        1. check to see if -k is already being used
+        2. if it isn't then add `-k uriOfSavedFile` to the args
+
+        */
+        if (fileUri !== undefined) {
+            // filter out arg "." if it exits
+            const filteredPytestArgs = pytestArgs.filter((arg) => arg !== '.');
+            filteredPytestArgs.push(fileUri.fsPath);
```

Comment:
Thanks for putting it together. I believe you also need to push "-k" at the beginning unless I misunderstood how this part works. But I think this only affects running tests, i.e. after collecting, so it probably doesn't affect performance. 
I have also seen that there are other options:
```
  --ignore=path         ignore path during collection (multi-allowed).
  --ignore-glob=path 
  --override-ini python_files (args):  glob-style file patterns for Python test module discovery
```

Maybe `python_files` is the one to use, but I also dont know if it will work since it may not support full file paths. 😞